src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -347,7 +347,7 @@ private sealed class Extensions<TExtension>
             private readonly bool _allowNetFramework;
             private ImmutableArray<TExtension> _lazyAllExtensions;
             private ImmutableDictionary<string, ImmutableArray<TExtension>> _lazyExtensionsPerLanguage;
-            private ImmutableDictionary<string, ImmutableHashSet<string>>? _lazyExtensionTypeNameMap;
+            private ImmutableSortedDictionary<string, ImmutableSortedSet<string>>? _lazyExtensionTypeNameMap;
```

Comment:
This is a map from language identifier to the set of analyzer / generator type names correct? Can this dictionary ever have more than two keys?

Reply:
Yes, the attribute is actually just a string, so can contain arbitrary values. In the generator tests we have a generator that targets F# to make sure its exercised correctly (and doesn't show up when you filter for CSharp or VB)

Reply:
It just reminds me of a performance quip I heard a few times. Essentially if you have less than ~10 items or so then you're better off just using an array vs. a map. Not a request for this PR more wanted to make sure I understood everything correctly. 

@sharwell FYI but doubt it meaningfully contributes to performance

Reply:
I don't believe this one has been showing up. Also, keep in mind that both `ImmutableDictionary<TKey, TValue>` and `ImmutableHashSet<T>` are sorted - as long as the equality comparer produces deterministic hash codes the result will be deterministic.